lib/util/http/index.ts

```diff
@@ -253,23 +256,29 @@ export class Http<Opts extends HttpOptions = HttpOptions> {
     if (body) {
       opts.json = body;
     }
-    const res = await this.request<T>(url, opts);
+    const res = await this.request<ResT>(url, opts);
+
+    if (!schema) {
+      return { ...res, body: res.body };
+    }
 
-    if (schema) {
-      match(schema, res.body, onSchemaError);
+    const parsed = await schema.safeParseAsync(res.body);
+    if (!parsed.success) {
+      logger.once.info({ err: parsed.error }, `Response does not match schema`);
+      return { ...res, body: res.body };
```

Comment:
see above, i don't see a benefit for creating a flat copy here.

Reply:
My guess is that we're caching GET responses in a way that same `HttpResponse` object is returned for each request hitting same cache key, and we want to avoid its mutability.

Reply:
Anyway, I don't like it too, but I think we need to address this in separate PR

Reply:
ok, but the extra body assignment really makes any sense, as it doesn't change anything on final object.

Reply:
Yes, it's actually doesn't help to prevent mutation, but still let's do this in separate PR just to make it easier to revert